File: app/src/main/java/com/library/list/singlechoice/SingleChoiceListActivity.java
package com.library.list.singlechoice;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.os.Bundle;
import android.view.View;

import com.blankj.utilcode.util.ToastUtils;
import com.common.base.BaseActivity;
import com.common.http.XRetrofit;
import com.common.http.base.BaseObserver;
import com.common.http.base.RxJavaHelper;
import com.common.http.bean.ExampleListBean;
import com.common.http.helper.Mobile;
import com.common.library.bravh_rvadapter.BaseRecyclerAdapter;
import com.common.weight.CommonRecyclerView;
import com.library.R;
import com.library.list.refreshload.RefreshLoadListAdapter;

import java.util.Collections;
import java.util.List;

/**
 * 单选列表Activity
 */
public class SingleChoiceListActivity extends BaseActivity {

    private CommonRecyclerView rvSingleChoice;
    private SingleChoiceListAdapter singleChoiceListAdapter;
    private final int page = 1;
    private final int pageNum = 20;

    @Override
    public int getLayout() {
        return R.layout.activity_single_choice_list;
    }

    @Override
    public void initViewIds() {
        rvSingleChoice = findViewById(R.id.rv_single_choice);
    }

    @Override
    public void initView() {
        initSingleChoiceRecycler();
    }

    private void initSingleChoiceRecycler() {
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        rvSingleChoice.setLayoutManager(layoutManager);
        singleChoiceListAdapter = new SingleChoiceListAdapter(Collections.emptyList());
        rvSingleChoice.setAdapter(singleChoiceListAdapter);
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        getList(page, pageNum);
    }

    @Override
    public void initListener() {
        super.initListener();
        singleChoiceListAdapter.setOnItemChildClickListener((adapter, view, position) -> {
            if (view.getId() == R.id.iv_select) {
                //点击按钮,将当前点击的下标传进去,刷新适配器
                singleChoiceListAdapter.setSelPos(position);
                singleChoiceListAdapter.notifyDataSetChanged();
            }
        });
    }

    /**
     * 获取集合
     *
     * @param page:    页数
     * @param pageNum: 条目数
     */
    private void getList(int page, int pageNum) {
        XRetrofit.getApi()
                .getList(Mobile.getList(page, pageNum))
                .compose(RxJavaHelper.observeOnMainThread())
                .subscribe(new BaseObserver<List<ExampleListBean>>() {
                    @Override
                    public void onSuccess(List<ExampleListBean> response) {
                        if (response == null)
                            return;
                        singleChoiceListAdapter.setNewData(response);
                    }
                });
    }

    /**
     * 获取选中的下标
     *
     * @param view
     */
    public void btnGetSingleData(View view) {
        if (singleChoiceListAdapter.getSelPos() == -1) {
            ToastUtils.showShort("你还未选中数据");
            return;
        }
        ToastUtils.showShort("选中的标题是---"
                + singleChoiceListAdapter.getData()
                .get(singleChoiceListAdapter.getSelPos()).getTitle());
    }
}

File: app/src/main/java/com/library/list/singlechoice/SingleChoiceListActivity.kt
package com.library.list.singlechoice

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.blankj.utilcode.util.ToastUtils
import com.common.base.BaseActivity
import com.common.http.XRetrofit.api
import com.common.http.base.BaseObserver
import com.common.http.base.RxJavaHelper.observeOnMainThread
import com.common.http.bean.ExampleListBean
import com.common.http.helper.Mobile
import com.common.library.bravh_rvadapter.BaseRecyclerAdapter
import com.common.library.bravh_rvadapter.BaseRecyclerAdapter.OnItemChildClickListener
import com.common.weight.CommonRecyclerView
import com.library.R
import kotlinx.android.synthetic.main.activity_single_choice_list.*

/**
 * 单选列表Activity
 */
class SingleChoiceListActivity : BaseActivity() {
    private var singleChoiceListAdapter: SingleChoiceListAdapter? = null
    private val page = 1
    private val pageNum = 20
    override val layout: Int
        get() = R.layout.activity_single_choice_list

    override fun initViewIds() {
    }

    override fun initView() {
        initSingleChoiceRecycler()
    }

    private fun initSingleChoiceRecycler() {
        val layoutManager = LinearLayoutManager(this)
        rv_single_choice!!.layoutManager = layoutManager
        singleChoiceListAdapter = SingleChoiceListAdapter(emptyList())
        rv_single_choice!!.adapter = singleChoiceListAdapter
    }

    override fun initData(savedInstanceState: Bundle?) {
        getList(page, pageNum)
    }

    override fun initListener() {
        super.initListener()
        singleChoiceListAdapter!!.onItemChildClickListener = OnItemChildClickListener { adapter: BaseRecyclerAdapter<*, *>?, view: View, position: Int ->
            if (view.id == R.id.iv_select) { //点击按钮,将当前点击的下标传进去,刷新适配器
                singleChoiceListAdapter!!.selPos = position
                singleChoiceListAdapter!!.notifyDataSetChanged()
            }
        }
    }

    /**
     * 获取集合
     *
     * @param page:    页数
     * @param pageNum: 条目数
     */
    private fun getList(page: Int, pageNum: Int) {
        api?.getList(Mobile.getList(page, pageNum))
                ?.compose(observeOnMainThread())
                ?.subscribe(object : BaseObserver<List<ExampleListBean>>() {
                    override fun onSuccess(response: List<ExampleListBean>) {
                        singleChoiceListAdapter!!.setNewData(response)
                    }
                })
    }

    /**
     * 获取选中的下标
     *
     * @param view
     */
    fun btnGetSingleData(view: View?) {
        if (singleChoiceListAdapter!!.selPos == -1) {
            ToastUtils.showShort("你还未选中数据")
            return
        }
        ToastUtils.showShort("选中的标题是---"
                + singleChoiceListAdapter!!.data[singleChoiceListAdapter!!.selPos].title)
    }
}